### PR TITLE
Feature/sutd assignment2

### PR DESCRIPTION
**Pull request by annagav**
Contains working:
- Western blot
- Flow Cytometry
- Microscopy

Includes modifications to the microscopy technique of Exercise 3.

### CODE REVIEW

**Comment by bdero** [comment]
Understanding that your work on SUTD Assignment 2 depends on some changes from the work on Exercise 3, could you also create a pull request for the Exercise 3 branch and reference that PR as a dependency for this PR? It'll make it a bit easier to review.

File: html_app/assignments/sutd_intro_biology_ps2/addmultipledialog.gss [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    width:115px;
+    margin-left: 10px;
+    margin-right:10px;
+    margin-top: 10px;
+
+}
+
+.sutd_ps2_button_float {
+    display:block;
+}
```
Comment (pdpinch) [inline html_app/assignments/sutd_intro_biology_ps2/addmultipledialog.gss:102]: Is it common for a new exercise to require 10 new style classes?

File: html_app/components/FACSModelFactory.js [hunk 2]
```diff
@@ -150,6 +150,24 @@ scb.components.FACSModelFactory = function scb_components_FACSModelFactory(model
 			function peak2Ug2(x){
 				  return normal_dist(x, 0.31, 0.14, -2, true)*6;
 			}
+
+            /*
+            *The following 3 functions describe the asynchronous image
+            * with a tall peak at 50 and a small bump at 100, and flat
+            * middle ground.
+            */
+            function bump100(x){
```
Comment (pdpinch) [inline html_app/components/FACSModelFactory.js:159]: For the future, this could have been handled as a separate issue + PR

**Comment by pdpinch** [comment]
Same comment/question as https://github.com/starteam/starcellbio_html/pull/35 -- how complete an implementation is this of SUTD #2?

File: html_app/assignments/sutd_intro_biology_ps2/addmultipledialog.gss (hunk 1 above)
Comment (annagav) [inline html_app/assignments/sutd_intro_biology_ps2/addmultipledialog.gss:102]: This is the way all previous assignments were built. It probably does not have be this way, unless in the assignment builder we let the instructor customize the "Add Sample" dialog.

File: html_app/assignments/sutd_intro_biology_ps2/instructions.soy [hunk 3]
```diff
@@ -0,0 +1,16 @@
+{namespace scb_assignment_sutd_intro_biology_ps2}
+
+/**
+
+*/
+{template .experiment_setup}
+<ul role='list'  class='scb_assignment_specific_tufts_experiment_setup'>
```
Comment (pdpinch) [inline html_app/assignments/sutd_intro_biology_ps2/instructions.soy:7]: Is it an issue that the class here appears to be from a different assignment? 'scb_assignment_specific_tufts_experiment_setup' ?

File: html_app/components/FACSModelFactory.js [hunk 4]
```diff
@@ -237,20 +274,24 @@ scb.components.FACSModelFactory = function scb_components_FACSModelFactory(model
...
+
 
-               // console.log("Ticks: "+tick1);
-                if (sum != 0) {
-                    _.each(data, function (s, index) {
-                        data[index][1] = data[index][1] / sum * (template.model.facs.max ? ((big_const*tick1)/template.model.facs.max)*number_of_curves: 2750  );
-                        console.log("y= "+data[index][1]);
-                    });
-                }
                 _.each(data, function (s, index) {
```
Comment (pdpinch) [inline html_app/components/FACSModelFactory.js]: Why does this method get called twice in a row?
`_.each(data, function (s, index)`

File: html_app/model/Treatment.js [hunk 5]
```diff
@@ -73,25 +73,34 @@ scb.Treatment = function scb_Treatment(data, context, parent) {
...
-
 			var hours = Math.floor((time % 86400) / 3600);
-			var minutes = Math.round((time % 3600) / 60);
-			
+			var minutes = Math.floor((time % 3600) / 60);
 			var months = Math.floor(time /2592000);
-			
 			var weeks = Math.floor((time % 2592000) / 604800);
-			var now = (time < 60 );
+			var now = (time < 30 );
```
Comment (pdpinch) [inline html_app/model/Treatment.js:81]: Wow. It's really confusing to have this code duplicated for each assignment. I opened issue https://github.com/starteam/starcellbio_html/issues/61 to think about refactoring it.

Comment (annagav) [inline html_app/model/Treatment.js:81]: This code is used by all assignments, it is not being duplicated.

Comment (pdpinch) [inline html_app/model/Treatment.js:81]: My bad. I was completely lost. 

But we still do have very similar code in https://github.com/annagav/starcellbio_html/blob/feature/sutd_assignment2/html_app/model/Treatment.js#L45-L64 and https://github.com/annagav/starcellbio_html/blob/feature/sutd_assignment2/html_app/model/Treatment.js#L45-L64

**Comment by pdpinch** [comment]
I think this needs a rebase. When I test this branch, I'm still seeing the old intro text, "Here come instructions when we build them"

https://github.com/annagav/starcellbio_html/blob/feature/sutd_assignment2/html_app/master_model.js#L1564

I'm hoping also that after a rebase the master model changes will be smaller and that I can diff them. Right now I can't see anything in the github.com ui.

**Comment by pdpinch** [comment]
Nevermind about the rebase.

**Pull request merged**